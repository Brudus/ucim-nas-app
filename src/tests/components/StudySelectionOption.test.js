import React from 'react';
import { shallow } from 'enzyme';
import StudySelectionOption from '../../components/StudySelectOption';

let calculateNewSchedule;

beforeEach(() => {
    calculateNewSchedule = jest.fn();
});

test('should render StudySelectionOption with text', () => {
    const grade = 0;
    const text = "Again"
    const wrapper = shallow(<StudySelectionOption grade={grade} text={text} calculateNewSchedule={calculateNewSchedule} />);
    expect(wrapper).toMatchSnapshot();
});

test('should call calculateNewSchedule with grade when button is pressed', () => {
    const grade = 3;
    const text = "Hard"
    const wrapper = shallow(<StudySelectionOption grade={grade} text={text} calculateNewSchedule={calculateNewSchedule} />);
    wrapper.find('button').simulate('click');
    expect(calculateNewSchedule).toHaveBeenLastCalledWith(grade);
});