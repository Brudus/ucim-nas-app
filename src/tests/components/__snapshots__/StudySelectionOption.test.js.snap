// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render StudySelectionOption with text 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Again
  </button>
</div>
`;
